File: Python/Book/B/T12/binarySearch.py
def binarySearch(T, x):
    l = 0
    r = len(T) - 1
    while l < r:
        m = int((l + r) / 2)
        if(T[m] == x):
            return m
        elif(T[m] > x):
            r = m - 1
        else:
            l = m + 1
    return -1

File: Search/binarySearch/binarySearch.c
#include "binarySearch.h"

key binarySearch(key *a, key x, key low, key high)
{
    key mid = (low + high) / 2;
    if (low > high)
    {
        return 0;
    }
    else{
        if (a[mid] == x)
        {
            return mid;
        }
        else if (a[mid] > x)
        {
            return binarySearch(a, x, low, mid - 1);
        }
        else
        {
            return binarySearch(a, x, mid + 1, high);
        }
    }
}